Added changelog for partner SDK

fivetran-abdulsalam opened the pull request:
Links https://fivetran.height.app/T-708746
Add changelog file and June release

manjutapali commented on CHANGELOG.md:
```diff
@@ -0,0 +1,24 @@
+**ConfigurationForm Changes**
+- We added options to provide default value and placeholder for the Field.
+- We Introduced  FieldSet  support, which allows to group the Fields and add visibility condition to control displaying on the form.
+- The FormField format is updated to support FieldSet. The single attribute should be used to assign the usual fields.
```
NIT: can be changed from `usual fields` to `regular fields`.

fivetran-abdulsalam replied:
incorporated

manjutapali commented on CHANGELOG.md:
```diff
@@ -0,0 +1,24 @@
+
+**SourceConnector Changes**
+- We removed LogEntry from Update operation. Now, partners can use STD::OUT to send logs. For more details, please refer to this.
+- We renamed few enums and services:
```
We have renamed the few enums and services

manjutapali commented on CHANGELOG.md:
```diff
@@ -0,0 +1,24 @@
+
+**DestinationConnector Changes**
+- Added NAIVE_TIME data type support.
+- We renamed the service definition from Destination to DestinationConnector.
```
We have renamed the service

fivetran-abdulsalam replied:
incorporated

manjutapali reviewed (commented):
added comments

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,29 @@
+- The FormField format is updated to support FieldSet. The single attribute should be used to assign the regular fields.
+
+**SourceConnector Changes**
+- We removed `LogEntry` from Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to [this](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging).
```
```suggestion
- We removed `LogEntry` from Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
```

fivetran-abdulsalam replied:
Incorporated

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,29 @@
+- Added support for `NAIVE_TIME` datatype. Partners can now send data with this datatype.
+
+**DestinationConnector Changes**
+- Added `NAIVE_TIME` data type support.
```
```suggestion
```

fivetran-abdulsalam replied:
Incorporated

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,29 @@
+- We have renamed the few enums and services:
+  - Renamed the service definition from `Connector` to `SourceConnector`.
+  - Renamed the `OpType` field to `RecordType`.
+- Added support for `NAIVE_TIME` datatype. Partners can now send data with this datatype.
```
```suggestion
```

fivetran-abdulsalam replied:
Incorporated

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,29 @@
+
+**SourceConnector Changes**
+- We removed `LogEntry` from Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to [this](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging).
+- We have renamed the few enums and services:
```
```suggestion
```

fivetran-abdulsalam replied:
Incorporated

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,29 @@
+**SourceConnector Changes**
+- We removed `LogEntry` from Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to [this](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging).
+- We have renamed the few enums and services:
+  - Renamed the service definition from `Connector` to `SourceConnector`.
```
```suggestion
- Renamed service definition from `Connector` to `SourceConnector`.
```

fivetran-abdulsalam replied:
Incorporated

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,29 @@
+- We removed `LogEntry` from Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to [this](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging).
+- We have renamed the few enums and services:
+  - Renamed the service definition from `Connector` to `SourceConnector`.
+  - Renamed the `OpType` field to `RecordType`.
```
```suggestion
- Renamed `OpType` to `RecordType`.
```

fivetran-abdulsalam replied:
Incorporated

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,29 @@
+**DestinationConnector Changes**
+- Added `NAIVE_TIME` data type support.
+- We have renamed the service definition from `Destination` to `DestinationConnector`.
+- A new rpc method `Capabilities` is introduced. This brings the functionalities:
```
```suggestion
- A new rpc method `Capabilities` is introduced. This brings the following functionalities:
```

fivetran-abdulsalam replied:
Incorporated

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,29 @@
+- Added `NAIVE_TIME` data type support.
+- We have renamed the service definition from `Destination` to `DestinationConnector`.
+- A new rpc method `Capabilities` is introduced. This brings the functionalities:
+    - Adjust column types: Supports adjusting Partner data type for each Fivetran type.
```
```suggestion
    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
```

fivetran-abdulsalam replied:
Incorporated

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,29 @@
+#### June 2024
+
+-----------
```
```suggestion
-----------
+ - Added a new `NAIVE_TIME` data type.

```

fivetran-abdulsalam replied:
Incorporated

5tran-alexil commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+    - Introduced a new field `supports_history_mode`, which indicates whether the destination supports history mode.
+- Changes in AlterTableRequest:
+    - Takes `table_name` instead of `Table`
+    - Introduced SchemaDiff field, which supports adding new columns, change column type, and update primary keys.  
```
```suggestion
    - Introduced a new field, `SchemaDiff`, which supports adding new columns, change column type, and update primary keys.  
```

5tran-alexil commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+
+**ConfigurationForm Changes**
+- We added options to provide default value and placeholder for the Field.
+- We Introduced FieldSet support, which allows to group the Fields and add visibility condition to control displaying on the form.
```
```suggestion
- We introduced FieldSet support, which allows to group the Fields and add visibility condition to control displaying on the form.
```

5tran-alexil commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
+    - Define the max value supported for the columns types.
+    - We have added support for providing parquet batch files. Now, partner can choose the file type to create batch files.
+    - Introduced a new field `supports_history_mode`, which indicates whether the destination supports history mode.
```
```suggestion
    - Introduced a new field, `supports_history_mode`, which indicates whether the destination supports history mode.
```

5tran-alexil commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+- We have renamed the service definition from `Destination` to `DestinationConnector`.
+- A new rpc method `Capabilities` is introduced. This brings the following functionalities:
+    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
+    - Define the max value supported for the columns types.
```
```suggestion
    - Define the maximal value supported for the column types.
```

5tran-alexil commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+- Added a new `NAIVE_TIME` data type.
+
+**ConfigurationForm Changes**
+- We added options to provide default value and placeholder for the Field.
```
```suggestion
- We added options to provide default value and placeholder for `Field`.
```

5tran-alexil commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+**ConfigurationForm Changes**
+- We added options to provide default value and placeholder for the Field.
+- We Introduced FieldSet support, which allows to group the Fields and add visibility condition to control displaying on the form.
+- The FormField format is updated to support FieldSet. The single attribute should be used to assign the regular fields.
```
```suggestion
- The `FormField` format is updated to support `FieldSet`. A single attribute should be used to assign the regular fields.
```

5tran-alexil commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+    - Introduced SchemaDiff field, which supports adding new columns, change column type, and update primary keys.  
+- Changes in WriteBatchRequest:
+  - We have added support for parquet batch files configuration details.
+  - Introduced new field `HistoryMode`, which indicates history mode is enabled for table.
```
```suggestion
  - We have introduced new field `HistoryMode`, which indicates that history mode is enabled for the table.
```

ediril replied:
This is not how changelogs are written from what I've seen on the internet.. no need to use present perfect tense, past tense is sufficient. Also no need to specify "we".

5tran-alexil commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+    - We have added support for providing parquet batch files. Now, partner can choose the file type to create batch files.
+    - Introduced a new field `supports_history_mode`, which indicates whether the destination supports history mode.
+- Changes in AlterTableRequest:
+    - Takes `table_name` instead of `Table`
```
```suggestion
    - It now uses `table_name` instead of `Table`.
```

5tran-alexil commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+    - Define the max value supported for the columns types.
+    - We have added support for providing parquet batch files. Now, partner can choose the file type to create batch files.
+    - Introduced a new field `supports_history_mode`, which indicates whether the destination supports history mode.
+- Changes in AlterTableRequest:
```
```suggestion
- Changes in `AlterTableRequest`:
```

5tran-alexil commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+**SourceConnector Changes**
+- We removed `LogEntry` from Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
+- Renamed service definition from `Connector` to `SourceConnector`.
+- Renamed `OpType` to `RecordType`.
```
```suggestion
- We have renamed `OpType` to `RecordType`.
```

5tran-alexil commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+- The FormField format is updated to support FieldSet. The single attribute should be used to assign the regular fields.
+
+**SourceConnector Changes**
+- We removed `LogEntry` from Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
```
```suggestion
- We have removed `LogEntry` from the Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
```

fivetran-abdulsalam commented:
Incorporated all the comments

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+
+**ConfigurationForm Changes**
+- We have added options to provide default value and placeholder for `Field`.
+- We have introduced `FieldSet` support, which allows to group the `Fields` and add visibility condition to control displaying on the form.
```
```suggestion
- Introduced `FieldSet` support in `FormField`, which allows to group the `Fields` and add visibility condition to control displaying on the form.
```

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+**ConfigurationForm Changes**
+- We have added options to provide default value and placeholder for `Field`.
+- We have introduced `FieldSet` support, which allows to group the `Fields` and add visibility condition to control displaying on the form.
+- The `FormField` format is updated to support `FieldSet`. A single attribute should be used to assign the regular fields.
```
```suggestion
```

fivetran-abdulsalam replied:
Incorporated

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+- Added a new `NAIVE_TIME` data type.
+
+**ConfigurationForm Changes**
+- We have added options to provide default value and placeholder for `Field`.
```
```suggestion
- Added options to provide default value and placeholder for `Field`.
```

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+- The `FormField` format is updated to support `FieldSet`. A single attribute should be used to assign the regular fields.
+
+**SourceConnector Changes**
+- We have removed `LogEntry` from the Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
```
```suggestion
- Removed `LogEntry` from the Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
```

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+
+**SourceConnector Changes**
+- We have removed `LogEntry` from the Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
+- We have renamed service definition from `Connector` to `SourceConnector`.
```
```suggestion
- Renamed service definition from `Connector` to `SourceConnector`.
```

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+**SourceConnector Changes**
+- We have removed `LogEntry` from the Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
+- We have renamed service definition from `Connector` to `SourceConnector`.
+- We have renamed `OpType` to `RecordType`.
```
```suggestion
- Renamed `OpType` to `RecordType`.
```

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+- We have renamed `OpType` to `RecordType`.
+
+**DestinationConnector Changes**
+- We have renamed the service definition from `Destination` to `DestinationConnector`.
```
```suggestion
- Renamed the service definition from `Destination` to `DestinationConnector`.
```

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+
+**DestinationConnector Changes**
+- We have renamed the service definition from `Destination` to `DestinationConnector`.
+- A new rpc method `Capabilities` is introduced. This brings the following functionalities:
```
```suggestion
- Introduced a new rpc method `Capabilities` which brings the following functionalities:
```

fivetran-abdulsalam replied:
Incorporated

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+- We have renamed the service definition from `Destination` to `DestinationConnector`.
+- A new rpc method `Capabilities` is introduced. This brings the following functionalities:
+    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
+    - Define the maximal value supported for the column types.
```
```suggestion
    - Define maximum value supported by the destination column types.
```

fivetran-abdulsalam replied:
Incorporated

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+- A new rpc method `Capabilities` is introduced. This brings the following functionalities:
+    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
+    - Define the maximal value supported for the column types.
+    - We have added support for providing parquet batch files. Now, partner can choose the file type to create batch files.
```
```suggestion
    - Added support for parquet batch files.
```

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
+    - Define the maximal value supported for the column types.
+    - We have added support for providing parquet batch files. Now, partner can choose the file type to create batch files.
+    - We have introduced a new field, `supports_history_mode`, which indicates whether the destination supports history mode.
```
```suggestion
    - Introduced a new field, `supports_history_mode`, which indicates whether the destination supports history mode.
```

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+    - We have introduced a new field, `supports_history_mode`, which indicates whether the destination supports history mode.
+- Changes in `AlterTableRequest`:
+    - It now uses `table_name` instead of `Table`
+    - We have introduced a new field, `SchemaDiff`, which supports adding new columns, changing column type, and updating primary keys.  
```
```suggestion
    - Introduced a new field, `SchemaDiff`, which supports adding new columns, changing column type, and updating primary keys.  
```

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+    - It now uses `table_name` instead of `Table`
+    - We have introduced a new field, `SchemaDiff`, which supports adding new columns, changing column type, and updating primary keys.  
+- Changes in WriteBatchRequest:
+  - We have added support for parquet batch files configuration details.
```
```suggestion
  - Added support for parquet batch files configuration details.
```

ediril commented on CHANGELOG.md:
```diff
@@ -0,0 +1,28 @@
+    - We have introduced a new field, `SchemaDiff`, which supports adding new columns, changing column type, and updating primary keys.  
+- Changes in WriteBatchRequest:
+  - We have added support for parquet batch files configuration details.
+  - We have introduced new field `HistoryMode`, which indicates that history mode is enabled for the table.
```
```suggestion
  - Introduced a new field `HistoryMode`, which indicates that history mode is enabled for the table.
```

fivetran-abdulsalam commented:
@ediril, As per @5tran-alexil  suggestions I am changes from `We have introduced` to `Introduced` and from `We have added` to `Added`. Please check this comment: https://github.com/fivetran/fivetran_sdk/pull/44#discussion_r1644454002

5tran-alexil reviewed (approved):
LGTM

fivetran-abdulsalam commented:
@ediril Could you please re-review this PR

5tran-alexil commented:
@fivetran-abdulsalam Is there something that blocks from merging this changelog?

fivetran-abdulsalam commented:
Team has decided not to go with the changes and may go few changes in iteration. This PR is not needed now, so closing it
